test/test_polygon.py

```diff
@@ -124,20 +125,213 @@ def test_copy_invalid_args(self):
+
+    def test_center_x__del(self):
+        """Ensures that x component cannot be deleted."""
+        poly = Polygon(_some_vertices.copy())
+        with self.assertRaises(AttributeError):
+            del poly.c_x
+
+    def test_center_y(self):
+        """Makes sure changing center y component does change the positions of vertices properly."""
+        vertices = _some_vertices.copy()
+        poly = Polygon(vertices)
+        poly_c_y = poly.c_y
+        poly.c_y = 100.0
+        for vertex in vertices:
+            vertex[1] += 100.0 - poly_c_y
+
+        vertices = [tuple(vertex) for vertex in vertices]
+        self.assertEqual(poly.vertices, vertices)
+
+    def test_center_y__invalid_value(self):
+        """Ensures the function can handle the polygon center y component by invalid data types."""
+        poly = Polygon(_some_vertices)
+        for value in (None, [], "1", (1,), [1, 2, 3]):
+            with self.assertRaises(TypeError):
+                poly.c_y = value
+
+    def test_center_y__del(self):
+        """Ensures that y component cannot be deleted."""
+        poly = Polygon(_some_vertices)
+        with self.assertRaises(AttributeError):
+            del poly.c_y
+
+    def test_center(self):
+        """Makes sure that setting new center moves the vertices properly."""
+        poly = Polygon(_some_vertices.copy())
+        poly_center = poly.center
+        poly.center = (200.0, 200.0)
+        vertices = _some_vertices.copy()
+        for vertex in vertices:
+            vertex[0] += 200.0 - poly_center[0]
+            vertex[1] += 200.0 - poly_center[1]
+
+        vertices = [tuple(vertex) for vertex in vertices]
+        self.assertEqual(poly.vertices, vertices)
+        self.assertEqual(poly.c_x, 200.0)
+        self.assertEqual(poly.c_y, 200.0)
+
+    def test_center__invalid_value(self):
+        """Ensures the function can handle the polygon center component by invalid data types."""
+        poly = Polygon(_some_vertices.copy())
+        for value in (None, [], "1", (1,), [1, 2, 3], (1, "s"), (None, 3), (2, (3,))):
+            with self.assertRaises(TypeError):
+                poly.center = value
+
+    def test_center__del(self):
+        """Ensures that center component cannot be deleted."""
+        poly = Polygon(_some_vertices.copy())
+        with self.assertRaises(AttributeError):
+            del poly.center
+
+    def test_move(self):
+        """Checks whether polygon moved correctly."""
+        poly = Polygon(_some_vertices.copy())
+        center_x = poly.c_x
+        center_y = poly.c_y
+
+        new_poly = poly.move(10.0, 10.0)
+        vertices = _some_vertices.copy()
+
+        for vertex in vertices:
+            vertex[0] += 10.0
+            vertex[1] += 10.0
+
+        vertices = [tuple(vertex) for vertex in vertices]
+        self.assertEqual(vertices, new_poly.vertices)
+        self.assertEqual(new_poly.c_x, center_x + 10.0)
+        self.assertEqual(new_poly.c_y, center_y + 10.0)
+
```

Comment:
I'd like one more test that ensures that using polygon.move on a polygon won't move the original polygon